src/notification.cpp

```diff
@@ -52,6 +56,142 @@ NotificationManagerProxy *notificationManager()
     return notificationManagerProxyInstance();
 }
 
+QString encodeDBusCall(const QString &service, const QString &path, const QString &iface, const QString &method, const QVariantList &arguments)
+{
+    QString s;
+
+    s.append(service).append(' ');
+    s.append(path).append(' ');
+    s.append(iface).append(' ');
+    s.append(method);
```

Comment:
This looks like a good place for `QStringBuilder`?

```
#include <QStringBuilder>

QString s = service % " " % path % " " % iface % " " % method;
```

See the section "More Efficient String Construction" in http://doc.qt.io/qt-5/qstring.html

Or at least reserve some space up front so that this method doesn't reallocate all the time when adding the arguments (http://doc.qt.io/qt-5/qstring.html#reserve)?

Or alternatively, put all elements in a `QStringList` and then use `::join()` on it to join elements with a space in one go?